File: Algorithm/Code/2.二叉树的遍历/二叉树的遍历.cpp
/*二叉树的遍历*/ 
/*1.先序遍历
  2.中序遍历
  3.后序遍历*/ 
#include<iostream>
#include<vector>
using namespace std;


//创建二叉树(使用数组进行模拟完全二叉树)  数组只可以模拟完全二叉树与其特殊的形态(满二叉树) 

vector<int> tree(1,0);  //创建长度为1的向量，并将其值设置为1，这样相当于直接将向量中的第一个值设置为0 
int n;

//先序遍历
void PreOrder(int m){
	if(m <= n){
		cout<<tree[m]<<"\t";
		PreOrder(2*m);
		PreOrder(2*m+1);
	}
}

//中序遍历
void InOrder(int m){
	if(m<=n){
		InOrder(2*m);
		cout<<tree[m]<<"\t";
		InOrder(2*m+1);
	}
}

void PostOrder(int m){
	if(m <= n){ //如果BT不为空
        PostOrder(2*m);    //递归调用左子树
        PostOrder(2*m+1);   //递归调用右子树
        printf("%d\t",tree[m]);   //访问节点
    }
}

int main(){
	int e;
	cin>>n;
//	tree.push_back(0);  //剔除0位置
	for(int i = 0;i<n;i++){
		//对完全二叉树进行赋值 
		cin>>e; 
		tree.push_back(e);
	}
	cout<<"前序遍历："<<endl; 
	PreOrder(1);
	puts("");
	cout<<"中序遍历："<<endl; 
	InOrder(1);
	puts(""); 
	cout<<"后序遍历："<<endl; 
	PostOrder(1);
	return 0;
} 

File: Algorithm/Code_Py/1.二叉树的遍历/二叉树的遍历.py
'''
二叉树的遍历:
  1.先序遍历
  2.中序遍历
  3.后序遍历
'''

import numpy as np
#使用数组模拟完全二叉树
#定义全局变量
n = 0
#1.创建空的数组
tree = np.zeros(1)

#先序遍历
def PreOrder(m):
    if m <= n:
        print(tree[m],end=" ")
        PreOrder(2*m)
        PreOrder(2*m+1)

#中序遍历
def InOrder(m):
    if m <= n:
        InOrder(2*m)
        print(tree[m],end=" ")
        InOrder(2*m+1)

#后序遍历
def PostOrder(m):
    if m <= n:
        PostOrder(2*m)
        PostOrder(2*m+1)
        print(tree[m],end=" ")






if __name__ == '__main__':
    n = int(input("请输入二叉树的结点个数"))
    for i in range(n):
        e = int(input("请对二叉树的结点进行赋值"))
        tree = np.append(tree,e)
    


    print("前序遍历")
    PreOrder(1)
    print("")
    print("中序遍历")
    InOrder(1)
    print("")
    print("后序遍历")
    PostOrder(1)